sandbox/cgroup/monitor.go

```diff
@@ -0,0 +1,152 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2022 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package cgroup
+
+import (
+	"path"
+
+	"github.com/snapcore/snapd/osutil"
+	"github.com/snapcore/snapd/osutil/inotify"
+)
+
+type inotifyWatcher struct {
+	wd        *inotify.Watcher
+	addWatch  chan *groupToWatch
+	groupList []*groupToWatch
+	pathList  map[string]int32
```

Comment:
It might be worth adding a comment that `groupList` and `pathList` are only accessed by the  `watcherMainLoop` goroutine, and that's why no locks are required to protect access to them.

Reply:
Done.